_posts/2023-01-19-torchwavelets/torchwavelets.Rmd

```diff
@@ -0,0 +1,613 @@
+
+Finally, the [Arctic Oscillation (AO)](https://en.wikipedia.org/wiki/Arctic_oscillation) is a ring-like pattern of sea-level pressure anomalies centered at the North Pole. (Its Southern-hemisphere equivalent is the [Antarctic Oscillation](https://en.wikipedia.org/wiki/Antarctic_oscillation).) AO's influence extends beyond the Arctic Circle, however; it is indicative of whether and how much Arctic air flows down into the middle latitudes. AO and NAO are strongly related, and might designate the same physical phenomenon at a fundamental level.
+
+Now, let's make these characterizations more concrete by looking at actual data.
+
+## Analysis: ENSO
+
+We begin with the best-known of these phenomena: ENSO. Data are available from 1854 onwards; however, for comparability with AO, we discard all records prior to January, 1950. For analysis, we pick `NINO34_MEAN`, the monthly average sea surface temperature in the Niño 3.4 region (i.e., the area between 5° South, 5° North, 190° East, and 240° East). Finally, we convert to a `tsibble`, the format expected by `feasts::STL()`.
+
+```{r}
+library(tidyverse)
+library(tsibble)
+
+download.file(
+  "https://bmcnoldy.rsmas.miami.edu/tropics/oni/ONI_NINO34_1854-2022.txt",
+  destfile = "ONI_NINO34_1854-2022.txt"
+)
+
+enso <- read_table("ONI_NINO34_1854-2022.txt", skip = 9) %>%
+  mutate(x = yearmonth(as.Date(paste0(YEAR, "-", `MON/MMM`, "-01")))) %>%
+  select(x, enso = NINO34_MEAN) %>%
+  filter(x >= yearmonth("1950-01"), x <= yearmonth("2022-09")) %>%
+  as_tsibble(index = x)
+
+enso
+```
+
+    # A tsibble: 873 x 2 [1M]
+              x  enso
+          <mth> <dbl>
+     1 1950 Jan  24.6
+     2 1950 Feb  25.1
+     3 1950 Mar  25.9
+     4 1950 Apr  26.3
+     5 1950 May  26.2
+     6 1950 Jun  26.5
+     7 1950 Jul  26.3
+     8 1950 Aug  25.9
+     9 1950 Sep  25.7
+    10 1950 Oct  25.7
+    # … with 863 more rows
+
+As already announced, we want to look at seasonal decomposition, as well. In terms of seasonal periodicity, what do we expect? Unless told otherwise, `feasts::STL()` will happily pick a window size for us. However, there'll likely be several important frequencies in the data. (Not wanting to ruin the suspense, but for AO and NAO, this will definitely be the case!). Besides, we want to compute the Fourier Transform anyway, so why not do that first?
+
+Here is the power spectrum:
+
+```{r}
+library(torch)
+fft <- torch_fft_fft(as.numeric(scale(enso$enso)))
+
+```
+
+In the below plot, the x axis corresponds to frequencies, expressed as "number of times per year". We only display frequencies up to and including the Nyquist frequency, i.e., half the sampling rate, which in our case is 12 (per year).
+
+```{r}
```

Comment:
You should add alt-text to your figures for accessibility. You can do this with knitr's `fig.alt` (discussed in [this blog post](https://www.rstudio.com/blog/knitr-fig-alt/)), or you can use [ggplot2 functionality](https://ggplot2.tidyverse.org/reference/labs.html) if it's ggplots.

Reply:
oh, thanks, I didn't think I'd ever forget this again, after doing it for the book (I found it quite hard, actually)